drv/hisi_comp.c

```diff
@@ -855,7 +855,7 @@ static int hisi_zip_init(struct wd_ctx_config_internal *config, void *priv)
 	handle_t h_ctx;
 	int i;
 
-	memcpy(&zip_ctx->config, config, sizeof(struct wd_ctx_config));
+	memcpy(&zip_ctx->config, config, sizeof(struct wd_ctx_config_internal));
```

Comment:
If wd_ctx_config_internal is changed , like add some struct member, it will casuse  segment fault when hisi_zip_exit

Reply:
If more struct members are appended in internal config, your instruction will only copy chunk data into internal config. If some struct members are removed from internal config (the size of internal config is less than the size of config), your instruction will trigger segment fault since some extra data is copied.

If more struct members are appended in internal config, original instruction will only copy limited data. If some struct members are removed from internal config (the size of internal config is less than the size of config), original instructionn will trigger segment fault since some extra data is copied.

So I suggest to add an extra check whether the size is OK.

Reply:
struct hisi_zip_ctx {
        struct wd_ctx_config_internal   config;
};

I found that zip_ctx->config is also type of wd_ctx_internal_config，so  there is a special reason to copy that sizeof wd_ctx_config? This memcpy is also used in sec and hpre init, I check their size is also sizeof(struct wd_ctx_config_internal)

Reply:
Could we change it to "struct wd_ctx_config"? If it's an internal structure, it shouldn't be accessed by test cases. It should only be accessed by UADK internally.

Reply:
wd_ctx_config is set by user, and wd_ctx_config_internal is used in side uadk, it is copy  from wd_ctx_config  when wd_comp_init, but two struct may be different, so I think use internal struct in driver level is reasonable.